lib/cloudregister/registerutils.py

```diff
@@ -240,37 +240,86 @@ def fetch_smt_data(cfg, proxies):
+        # infrastructure.
+        # As time passes is more likely that the framework completes routing
+        # setup for outbound traffic. Therefore we wait less time the longer
+        # we are in the process of attempting to access the update
+        # infrastructure.
+        # Processing delays are handled at two levels; the individual request
+        # timeout and the time we wait between the retry of obtaining the
+        # update server information. For each timeout the starting number
+        # 15 and 20 seconds, respectively is an arbitrary choice. The maximum
+        # wait time is dependent on the number of region servers in the
+        # framework. The maximum wait time is calculated as follows
+        #
+        #sumof(num_regions_srvs * 15/i) + sumof(20/j)
+        # where i = 1,2,3
+        #       j = 1,2
+        #
+        # For a configuration with 3 region servers the framework has 112.5
+        # seconds to setup the routing
+        #
+        max_attempts = 3
+        attempt = 0
+        while attempt < max_attempts:
+            retry_cnt = attempt + 1
+            request_timeout = 15/retry_cnt
+            retry_timeout = int(20/retry_cnt)
+            logging.info(
+                'Getting update server information, attempt %d' % retry_cnt
+            )
+            for srv in region_servers:
+                srvName = srv.strip()
+                logging.info('\tUsing region server: %s' % srvName)
+                certFile = cert_dir + '/' + srvName + '.pem'
+                if not os.path.isfile(certFile):
+                    logging.info(
+                        '\tNo cert found: %s skip this server' % certFile
+                    )
+                    continue
+                try:
+                    response = requests.get(
+                        'https://%s/%s' % (srvName, api),
+                        verify=certFile,
+                        timeout=request_timeout,
+                        proxies=proxies
+                    )
+                    if response.status_code == 200:
+                        attempt = max_attempts
+                        break
+                    else:
+                        logging.error('=' * 20)
+                        logging.error(
+                            'Server returned: %d' % response.status_code
+                        )
+                        logging.error('Server error: "%s"' % response.reason)
+                        logging.error('=' * 20)
+                        if srv == region_servers[-1]:
+                            logging.error('\tAll servers reported an error')
+                            if retry_cnt < max_attempts:
+                                log_msg = 'Waiting %d ' % retry_timeout
+                                log_msg += 'seconds before next attempt'
+                                logging.info(log_msg )
+                                time.sleep(retry_timeout)
+                                attempt += 1
```

Comment:
Hmm, in both places `attempt` increment is inside `if retry_cnt < max_attemps` branch.

So if `attempt = 2`, then `retry_cnt = attempt + 1 = 3`, then the condition `if retry_cnt < max_attemps` turns into `if 3 < 3`, which is false and that's an infinite loop again.

I would suggest `for attempt in range(max_attempts):` instead of a `while` loop.